src/installer/installer_test.go

```diff
@@ -1,16 +1,71 @@
 package installer
 
-//import (
-//	"github.com/eranco74/assisted-installer/src/config"
-//	"github.com/sirupsen/logrus"
-//)
-//
-//var _ = Describe("installer", func() {
-//	var conf = config.Config{Role: "master", ClusterID: "cluster-id", Device: "/dev/vda", S3EndpointURL: "https://s3.com", S3Bucket: "test"}
-//	var l = logrus.New()
-//	i := NewAssistedInstaller(l, conf)
-//
-//	It("master role happy flow", func() {
-//		i.InstallNode()
-//	})
-//})
+import (
+	"fmt"
+	"github.com/eranco74/assisted-installer/src/config"
+	"github.com/eranco74/assisted-installer/src/ops"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"testing"
+)
+
+func TestValidator(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "inventory_test")
+}
+
+var _ = Describe("installer", func() {
+	var conf = config.Config{Role: "master", ClusterID: "cluster-id", Device: "/dev/vda", S3EndpointURL: "https://s3.com", S3Bucket: "test"}
+	var l = logrus.New()
+	installDir := "/opt/install-dir"
+	device := "/dev/vda"
+	l.SetOutput(ioutil.Discard)
+	ctrl := gomock.NewController(GinkgoT())
+
+	mockops := ops.NewMockOps(ctrl)
+	i := NewAssistedInstaller(l, conf, mockops)
+
+	It("master role happy flow", func() {
+		mockops.EXPECT().Mkdir(installDir).Return(nil).Times(1)
+		fileUrl := "https://s3.com/test/cluster-id/master.ign"
+		ignPath := "/opt/install-dir/master.ign"
+		mockops.EXPECT().ExecPrivilegeCommand("curl", "-s", fileUrl, "-o", ignPath).Return("", nil).Times(1)
```

Comment:
if you are the same mock several times i suggest to make it a function. you don't have to pass any parameters if you define it something like this in the head of the test:
```
mockOpsExecPrivilegeCommandSuccess := func() {
mockops.EXPECT().ExecPrivilegeCommand("curl", "-s", fileUrl, "-o", ignPath).Return("", err).Times(1)
}
```
this way you have less places to change in case something change in the api.